### PR TITLE
build: only check --engine-strict for production deps

### PR DESCRIPTION
**Pull request by bcoe**
Perform an installation of our `--production` dependencies first, with `--engine-strict`, so that we can detect dependencies that have dropped support for our supported Node.js version.

### CODE REVIEW

**Review by alexander-fenster — APPROVED** [review]

**Review by JustinBeckwith — APPROVED** [review]

**Review by chingor13 — APPROVED** [review]

**Comment by bcoe** [comment]
going to test a little bit upstream before we merge.

**Comment by bcoe** [comment]
seems to be working well, and will unblock some repos.

**Pull request merged**
